src/testModel.ts

```diff
@@ -225,32 +236,48 @@ export class TestModel {
 
     if (created.length || deleted.length)
       await this._listFiles();
-    if (changed.length)
-      await this.listTests(changed);
-  }
-
-  async ensureTests(files: string[]) {
-    // Do not list tests if all files are already loaded, otherwise we
-    // end up with update loop when updating tests for visible editors.
-    let allFilesLoaded = true;
-    for (const project of this._projects.values()) {
-      const projectSuiteFiles = projectFiles(project);
-      for (const f of files) {
-        const fileSuite = projectSuiteFiles.get(f);
-        if (!fileSuite || (fileSuite as any)[listFilesFlag]) {
-          allFilesLoaded = false;
-          break;
-        }
-      }
-      if (!allFilesLoaded)
-        break;
+    if (changed.length) {
+      for (const c of changed)
+        this._filesWithListedTests.delete(c);
+      await this.ensureTests(changed);
```

Comment:
This makes us list tests for all changed files, e.g. on `git checkout`. Perhaps we should only re-list in those files that are already in the `_filesWithListedTests`?

Reply:
Let me do that.